client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx

```diff
@@ -165,12 +164,12 @@ const ProcessingStep: React.FC< ProcessingStepProps > = function ( props ) {
 	useEffect( () => {
 		if ( hasActionSuccessfullyRun && ! isSubmittedRef.current ) {
 			// We should only trigger signup completion for signup flows, so check if we have one.
-			if ( availableFlows[ flow ] ) {
-				availableFlows[ flow ]().then( ( flowExport ) => {
-					if ( flowExport.default.isSignupFlow ) {
-						recordSignupComplete( { ...destinationState } );
-					}
-				} );
+
+			if ( 'siteCreated' in destinationState ) {
+				const { siteCreated } = destinationState;
+				if ( siteCreated ) {
+					recordSignupComplete( { ...destinationState } );
+				}
```

Comment:
Nit: What do you think about refactoring this code a little so it's more concise?

```suggestion
			if ( destinationState.siteCreated ) {
			    recordSignupComplete( { ...destinationState } );
			}
```

Reply:
Oh, yes, definitely. I was trying to be excessively protective. 
Updated here 9e1fc6a46846ac36418582f3eb84e4999add4cd6